src/protocol/decoder.js

```diff
@@ -196,6 +203,23 @@ module.exports = class Decoder {
     return Decoder.decodeZigZag(result)
   }
 
+  // By default JavaScript's numbers are of type float64, performing bitwise operations converts the numbers to a signed 32-bit integer
+  // Unsigned Right Shift Operator >>> ensures the returned value is an unsigned 32-bit integer
+  readUVarInt() {
+    let currentByte
+    let result = 0
+    let i = 0
+    while (((currentByte = this.buffer[this.offset++]) & MOST_SIGNIFICANT_BIT) !== 0) {
+      result |= (currentByte & OTHER_BITS) << i
+      i += 7
+      if (i > 35) {
+        throw new Error('Variable length quantity is too long')
+      }
+    }
+    result |= currentByte << i
+    return result >>> 0
```

Comment:
While I do understand that the unsigned right shift is here to make sure that you round the number to a value between 0 and 2 ** 32 - 2, I don't really understand why it's necessary. Won't all valid unsigned varints fit in that range kind of by definition? And if somehow we get a number that's not in that range, is rounding the right thing to do? I really don't know.

Reply:
While running the boundaries test, the top boundary test would receive a `-1` instead of `4294967295`
`expect(decode32u(unsigned32(MAX_SAFE_UNSIGNED_INT))).toEqual(MAX_SAFE_UNSIGNED_INT)`

I was confused so I double checked it in Python and saw something was off. I looked it up online and saw a very detailed explaination to why that happens, to sum it up (straight from [here](https://stackoverflow.com/a/1822769)):

JS Numbers are double-precision floats, bitwise operators convert these numbers to 32-bit signed integers so anything above `2^31` would come back around as negative. To fix that I need a way to convert my result to fit unsigned integers. 

Luckily `>>> 0` converts the results of it's calculation to an unsigned integer and ensures my integer is between the range `0` and `0xFFFFFFFF`